bin/reference_data.py

```diff
@@ -1,142 +1,341 @@
+            payload = self.prepare_manual_metadata()
+            self.finish_metadata(payload)
+
+    @cached_property
+    def ftp(self):
+        ftp = ftplib.FTP('ftp.ensembl.org')
+        ftp.login()
+        ftp.cwd('pub')
+        return ftp
+
+    @cached_property
+    def all_ensembl_releases(self):
+        releases = []
+        for d in self.ftp.nlst():  # Ensembl doesn't support mlsd for some reason
+            if d.startswith('release-'):
+                releases.append(int(d.split('-')[1]))
+
+        releases.sort(reverse=True)
+        return ['release-%s' % r for r in releases]
+
+    @cached_property
+    def ensembl_release(self):
+        for release in self.all_ensembl_releases:
+            ls = self.ftp.nlst('%s/fasta/%s/dna' % (release, self.ftp_species))
+            top_level_fastas = [f for f in ls if f.endswith('dna.toplevel.fa.gz')]
+            if top_level_fastas and self.genome_version in top_level_fastas[0]:
+                return release
+
+        self.info('Could not find any Ensembl releases for ' + self.genome_version)
+
+    def latest_genome_version(self):
+        ls = self.ftp.nlst('%s/fasta/%s/dna' % (self.all_ensembl_releases[0], self.ftp_species))
+        if not ls:
+            raise DownloadError('Could not find %s in latest Ensembl release' % self.ftp_species)
+
+        ext = '.dna.toplevel.fa.gz'
+        top_level_fasta = [f for f in ls if f.endswith(ext)][0]
+        return '.'.join(top_level_fasta.split('.')[1:]).replace(ext, '')
+
+    def download_data(self):
+        self.info('Downloading reference genome')
+        base_dir = '%s/fasta/%s' % (self.ensembl_release, self.ftp_species)
+
+        ls = self.ftp.nlst(base_dir)
+        if os.path.join(base_dir, 'dna_index') in ls:
+            ls = self.ftp.nlst(os.path.join(base_dir, 'dna_index'))
+        else:
+            ls = self.ftp.nlst(os.path.join(base_dir, 'dna'))
+
+        files_to_download = [f for f in ls if 'dna.toplevel' in f]
+        self.info('Reference genome: found %i files, downloading %i', len(ls), len(files_to_download))
+        for f in files_to_download:
+            self.download_file(f)
+
+        ls = self.ftp.nlst('%s/variation/vcf/%s' % (self.ensembl_release, self.ftp_species))
+        files_to_download = [f for f in ls if '%s.vcf.gz' % self.ftp_species in f.lower()]
+        if not files_to_download:
+            i = input('%i files found. Enter a basename for a single VCF to use, or nothing to continue without '
+                      'variants. ' % len(ls))
+            if i:
+                files_to_download = [f for f in ls if i in f]  # this should include the index file
+            else:
+                files_to_download = []
+
+        self.info('Variation: found %i files, downloading %i', len(ls), len(files_to_download))
+        for f in files_to_download:
+            self.download_file(f)
+
+    def prepare_data(self):
```

Comment:
`prepare_data` could return the relative path of the files ready to be used in the payload

Reply:
This has now been done